crates/node-resource-metrics/src/collectors/process_metrics_collector.rs

```diff
@@ -177,17 +185,108 @@ impl Collector for ProcessMetricsCollector {
     }
 }
 
+/// A Collector for exposing Linux process metrics
+pub(crate) struct LinuxProcessMetricsCollector {
+    vm_size_bytes: Desc,
+    vm_rss_bytes: Desc,
+}
+
+impl LinuxProcessMetricsCollector {
+    fn new() -> Self {
+        let vm_size_bytes = Opts::new(LINUX_VMEM_SIZE, "Memory usage in bytes.")
```

Comment:
Can you clarify how is it different from memory metrics collected as a part of `ProcessMetricsCollector` ?

Reply:
Hmm, we may not need this info after all. The resident set is nothing but the memory usage above and virtual memory is already there. Anyhow, I wasn't able to get the working set size without manual computation, so we may not need to gather this at all.